Sources/TSCUtility/Version.swift

```diff
@@ -45,12 +45,122 @@ public struct Version: Hashable {
     }
 }
 
-extension Version: Comparable {
+/// An error that occurs during the creation of a version.
+public enum VersionError: Error, CustomStringConvertible {
```

Comment:
Would this be better as a nested type in `Version`?

Reply:
I tend to prefer the non-nested approach that you use here, because sometimes it makes sense to associate errors with a subsystem and not just a single type, and it seems better to be consistent than sometimes use nested errors and sometimes not.